### PR TITLE
Added info for getting compilation output from meson on autorebuild

### PR DESCRIPTION
**Pull request by story645**
## PR summary

Added a short sentence/command for building with compilation logs enabled cause I figure it's gonna be a commonish need of anyone working with the c-extension files. 

## PR checklist

- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
Personally, I think the env-var makes a lot more sense for the "I occasionally want to see the logs" pattern rather than setting it on install.

I'm also slightly weary of the "relatively recently" language here, as I don't particularly like putting time-sensitive verbiage in these docs (at least not in the past tense, less worried about things that "will change" and thus the docs need to be updated once it "has changed"), preferring the phrasing which does not make us have to either a) make a judgement call of when it should be removed before it is just untrue to call it "recent" or b) leave it untouched long enough for it to be unequivocally false. The phrasing of "before this commit" will remain true even if there is still a time where it is functionally irrelevant.

File: doc/devel/development_setup.rst [hunk 1]
```diff
@@ -164,12 +164,23 @@ command ::
...
 so that Python will be able to import Matplotlib from your development source directory.
-This allows you to import your modified version of Matplotlib without re-installing after
-every change. Note that before the merging of the `Meson port
-<https://github.com/matplotlib/matplotlib/pull/26621>`_, this is only true for ``*.py``
-files. If you change the C-extension source based on a commit before the change to the
-Meson build system (which might also happen if you change branches), you will have to
-re-run the above command.
+This allows you to import your modified version of Matplotlib without having to
+re-install after changing a ``.py`` or compiled extension file. On Windows, you
+may need to :external+meson-python:ref:`vsenv-example`.
```
Comment (tacaswell) [inline doc/devel/development_setup.rst:169]: Why is this generating warnings?

Comment (QuLogic) [reply doc/devel/development_setup.rst:169]: I'm not sure if that's a usual thing right now. On my current-version Git-for-Windows, Git Bash is in MSYS2, but there are no compilers installed.

Comment (QuLogic) [reply doc/devel/development_setup.rst]: The move to troubleshooting seems good.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**